initramfs-init.in

```diff
@@ -279,6 +279,22 @@ is_url() {
 	esac
 }
 
+# Do some tasks to make sure mounting the ZFS pool is A-OK
+prepare_zfs_root() {
+	local _root_vol=${KOPT_root#ZFS=}
+	local _root_pool=${_rool_vol%%/*}
+
+	# Force import if this has been imported on a different system previously
+	if [ "$KOPT_zfs_force" = 1 ]; then
+		zpool import -N -f $_root_vol
+	fi
+
+	# Ask for encryption password
+	if [ $(zpool list -H -o feature@encryption $_root_pool) = "active" ]; then
+		eval zfs load-key $_root_pool
```

Comment:
You've missed my previous message.
Say you have 2 datasets: `zroot/encrypted` and `zroot/rootfs`, first is encrypted, second is not
load-key command will actually execute in this case and return an error `Key load error: 'zroot/rootfs' is not encrypted.`

Reply:
`_root_pool` isn't the dataset but the pool though, so it's `zroot` not `zroot/...`

Reply:
`zpool` commands operates on pools.
`zfs` operates on datasets.
In this case zroot is a dataset name.

Reply:
Yes, but calling `zfs load-key <pool-name>` is vali,d, in this case `poolname` being zroot since the root of the database has the same name as the pool.

Reply:
I'm talking about the case when zroot (mountpoint=/) is unencrypted, but zroot/encrypted (mountpoint=/encrypted) is.
Current code will lead to spurious error on each boot for such users and it's a regression.

Reply:
Huh, you can mount `zroot`too? My bad, had expected you could only mount `zroot/vol/`.

Reply:
Also, I'm pretty sure you need encryption enabled on the root dataset in order to enable encryption for childrens.